ibm/data_source_ibm_is_instance_network_interface.go

```diff
@@ -0,0 +1,448 @@
+							Computed:    true,
+							Description: "The URL for this floating IP.",
+						},
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier for this floating IP.",
+						},
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique user-defined name for this floating IP.",
+						},
+					},
+				},
+			},
+			"href": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL for this network interface.",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The user-defined name for this network interface.",
+			},
+			"port_speed": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The network interface port speed in Mbps.",
+			},
+			"primary_ipv4_address": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The primary IPv4 address.",
+			},
+			"resource_type": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource type.",
+			},
+			"security_groups": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Collection of security groups.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"crn": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The security group's CRN.",
+						},
+						"deleted": &schema.Schema{
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "If present, this property indicates the referenced resource has been deleted and providessome supplementary information.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"more_info": &schema.Schema{
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Link to documentation about deleted resources.",
+									},
+								},
+							},
+						},
+						"href": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The security group's canonical URL.",
+						},
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier for this security group.",
+						},
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The user-defined name for this security group. Names must be unique within the VPC the security group resides in.",
+						},
+					},
+				},
+			},
+			"status": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the network interface.",
+			},
+			"subnet": &schema.Schema{
```

Comment:
subnet, as it contains information about the only subnet it is associated with. It is a typelist because it contains the CRN, ID and HREF etc of the same subnet